src/passes/OptimizeInstructions.cpp

```diff
@@ -163,7 +296,10 @@ Index getMaxBits(Expression* curr, LocalInfoProvider* localInfoProvider) {
       case EqZInt64:
         return 1;
       case WrapInt64:
+      case ExtendUInt32:
         return std::min(Index(32), getMaxBits(unary->value, localInfoProvider));
+      case ExtendSInt32:
+        return std::min(Index(64), getMaxBits(unary->value, localInfoProvider));
```

Comment:
this doesn't look right. the `min` operation doesn't do anything since 64 is the maximum value possible anyhow?

I think it might be `getMaxBits(value) == 32 ? 64 : getMaxBits(value)`

Reply:
yes, you're right!